syntax = "proto3";

package hipsterbrown.component.rgb.v1;

import "google/api/annotations.proto";

service RgbService {
  rpc Animate(AnimateRequest) returns (AnimateResponse) {
    option (google.api.http) = {
      post: "/hipsterbrown/api/v1/component/rgb/{name}/animate"
    };
  }

  rpc Fill(FillRequest) returns (FillResponse) {
    option (google.api.http) = {
      post: "/hipsterbrown/api/v1/component/rgb/{name}/fill"
    };
  }

  rpc Clear(ClearRequest) returns (ClearResponse) {
    option (google.api.http) = {
      post: "/hipsterbrown/api/v1/component/rgb/{name}/clear"
    };
  }

  rpc Stop(StopRequest) returns (StopResponse) {
    option (google.api.http) = {
      post: "/hipsterbrown/api/v1/component/rgb/{name}/stop"
    };
  }
}

message AnimateRequest {
  string name = 1;
}

message AnimateResponse {
  string text = 1;
}

message FillRequest {
  string name = 1;
  int32 red = 2;
  int32 green = 3;
  int32 blue = 4;
}

message FillResponse {
  string text = 1;
}

message ClearRequest {
  string name = 1;
}

message ClearResponse {
  string text = 1;
}
message StopRequest {
  string name = 1;
}

message StopResponse {
  string text = 1;
}
